<launch>

    <!-- To have the localhost machine defined --> 
    <include file="$(find metal)/launch/machines.launch"/>

    <arg name="executor" default="effibot"/>
    <arg name="vnet" default="true"/>
    <arg name="name" default="effibot3"/>
    <arg name="machine" default="localhost"/>

    <node name="metal" pkg="metal" type="hidden-ros.py" ns="$(arg name)" machine="$(arg machine)"
    		args="--executor=$(arg executor) --agentName=$(arg name) --waitSignal" >
        <remap from="hidden/repair/in" to="/hidden/repair" unless="$(arg vnet)"/>
        <remap from="hidden/repair/out" to="/hidden/repair" unless="$(arg vnet)"/>
        <remap from="hidden/communicate/in" to="/hidden/communicate" unless="$(arg vnet)"/>
        <remap from="hidden/communicate/out" to="/hidden/communicate" unless="$(arg vnet)"/>
        <remap from="hidden/mastnUpdate/in" to="/hidden/mastnUpdate" unless="$(arg vnet)"/>
        <remap from="hidden/mastnUpdate/out" to="/hidden/mastnUpdate" unless="$(arg vnet)"/>
    </node>

</launch>
